libtrap/src/ifc_tls.c

```diff
@@ -1022,18 +1015,24 @@ static int client_socket_connect(tls_receiver_private_t *c, struct timeval *tv)
       }
       break;
    }
+   /* there was no successfull connection for whole servinfo struct */
+   if (p == NULL) {
+      VERBOSE(CL_VERBOSE_LIBRARY, "recv client: Connection failed.");
+      rv = TRAP_E_TIMEOUT;
+   }
+
+   /* catching all possible errors from setting up socket before atempting tls handshake */
+   if (rv != TRAP_E_OK) {
+      CHECK_AND_FREE(servinfo, freeaddrinfo);
+      return rv;
+   }
 
    if (p != NULL) {
       inet_ntop(p->ai_family, get_in_addr((struct sockaddr *)p->ai_addr), s, sizeof s);
       VERBOSE(CL_VERBOSE_LIBRARY, "recv client: connected to %s", s);
    }
    CHECK_AND_FREE(servinfo, freeaddrinfo); /* all done with this structure */
 
-   if (p == NULL) {
-      VERBOSE(CL_VERBOSE_LIBRARY, "recv client: Connection failed.");
-      return TRAP_E_TIMEOUT;
-   }
-
    c->sd = sockfd;
    c->ssl = SSL_new(c->sslctx);
    SSL_set_fd(c->ssl, c->sd);
```

Comment:
This comment is for Line 1109: `return TRAP_E_OK` -> probably the same issue as in ifc_tcpip.c

Reply:
Here I think that TRAP_E_OK is ok. Because in case of TLS if we get here, there really was no problem.

I check for problems in socket setup here: https://github.com/jaroslavh/Nemea-Framework/blob/master/libtrap/src/ifc_tls.c#L1025

On lines below, I did not come up with better solution than return on the spot, because I need to stop before the negotiation. What I mean are lines 1057 and 1067.